test/test_gromov.py

```diff
@@ -1425,24 +1425,40 @@ def test_fgw_barycenter(nx):
     init_C /= init_C.max()
     init_Cb = nx.from_numpy(init_C)
 
-    Xb, Cb = ot.gromov.fgw_barycenters(
-        n_samples, [ysb, ytb], [C1b, C2b], ps=[p1b, p2b], lambdas=None,
-        alpha=0.5, fixed_structure=True, init_C=init_Cb, fixed_features=False,
-        p=None, loss_fun='square_loss', max_iter=100, tol=1e-3
-    )
+    try:  # to raise warning when `fixed_structure=True`and `init_C=None`

```

Comment:
to check that an erreor is Raised use : 
```suggestion
    with pytest.raises(ot.utils.UndefinedParameter):
```